File: japiwdbe/WdbeQFilList.java
/**
  * \file WdbeQFilList.h
  * Java API code for record of table TblWdbeQFilList
  * \author Alexander Wirthmueller
  * \date created: 11 Jul 2020
  * \date modified: 11 Jul 2020
  */

package apiwdbe;

import org.w3c.dom.*;
import sbecore.*;

public class WdbeQFilList {
	
	public WdbeQFilList(
				int jnum
				, String stubGrp
				, String stubOwn
				, String Filename
				, String srefRefIxVTbl
				, String titRefIxVTbl
				, String stubRefUref
				, String osrefKContent
				, String titOsrefKContent
				, String srefKMimetype
				, String titSrefKMimetype
				, int Size
			) {
		this.jnum = jnum;
		this.stubGrp = stubGrp;
		this.stubOwn = stubOwn;
		this.Filename = Filename;
		this.srefRefIxVTbl = srefRefIxVTbl;
		this.titRefIxVTbl = titRefIxVTbl;
		this.stubRefUref = stubRefUref;
		this.osrefKContent = osrefKContent;
		this.titOsrefKContent = titOsrefKContent;
		this.srefKMimetype = srefKMimetype;
		this.titSrefKMimetype = titSrefKMimetype;
		this.Size = Size;
	};

	public int jnum;
	public String stubGrp;
	public String stubOwn;
	public String Filename;
	public String srefRefIxVTbl;
	public String titRefIxVTbl;
	public String stubRefUref;
	public String osrefKContent;
	public String titOsrefKContent;
	public String srefKMimetype;
	public String titSrefKMimetype;
	public int Size;
	
	public boolean readXML(
				Document doc
				, String basexpath
				, boolean addbasetag
			) {
		if (addbasetag) basexpath = Xmlio.checkUclcXPaths(doc, basexpath, "WdbeQFilList");

		if (Xmlio.checkXPath(doc, basexpath)) {
			stubGrp = Xmlio.extractStringUclc(doc, basexpath, "stubGrp", "grp", null, 0);
			stubOwn = Xmlio.extractStringUclc(doc, basexpath, "stubOwn", "own", null, 0);
			Filename = Xmlio.extractStringUclc(doc, basexpath, "Filename", "fnm", null, 0);
			srefRefIxVTbl = Xmlio.extractStringUclc(doc, basexpath, "srefRefIxVTbl", "ret", null, 0);
			titRefIxVTbl = Xmlio.extractStringUclc(doc, basexpath, "titRefIxVTbl", "ret2", null, 0);
			stubRefUref = Xmlio.extractStringUclc(doc, basexpath, "stubRefUref", "reu", null, 0);
			osrefKContent = Xmlio.extractStringUclc(doc, basexpath, "osrefKContent", "cnt", null, 0);
			titOsrefKContent = Xmlio.extractStringUclc(doc, basexpath, "titOsrefKContent", "cnt2", null, 0);
			srefKMimetype = Xmlio.extractStringUclc(doc, basexpath, "srefKMimetype", "mim", null, 0);
			titSrefKMimetype = Xmlio.extractStringUclc(doc, basexpath, "titSrefKMimetype", "mim2", null, 0);
			Size = Xmlio.extractIntegerUclc(doc, basexpath, "Size", "siz", null, 0);
			
			return true;
		};

		return false;
	};
};

File: apiwdbe/WdbeQFilList.cpp
/**
	* \file WdbeQFilList.cpp
	* API code for table TblWdbeQFilList (implementation)
	* \copyright (C) 2016-2020 MPSI Technologies GmbH
	* \author Alexander Wirthmueller (auto-generation)
	* \date created: 5 Dec 2020
  */
// IP header --- ABOVE

#include "WdbeQFilList.h"

using namespace std;
using namespace Sbecore;
using namespace Xmlio;

/******************************************************************************
 class WdbeQFilList
 ******************************************************************************/

WdbeQFilList::WdbeQFilList(
			const uint jnum
			, const string stubGrp
			, const string stubOwn
			, const string Filename
			, const string srefRefIxVTbl
			, const string titRefIxVTbl
			, const string stubRefUref
			, const string osrefKContent
			, const string titOsrefKContent
			, const string srefKMimetype
			, const string titSrefKMimetype
			, const usmallint Size
		) {
	this->jnum = jnum;
	this->stubGrp = stubGrp;
	this->stubOwn = stubOwn;
	this->Filename = Filename;
	this->srefRefIxVTbl = srefRefIxVTbl;
	this->titRefIxVTbl = titRefIxVTbl;
	this->stubRefUref = stubRefUref;
	this->osrefKContent = osrefKContent;
	this->titOsrefKContent = titOsrefKContent;
	this->srefKMimetype = srefKMimetype;
	this->titSrefKMimetype = titSrefKMimetype;
	this->Size = Size;
};

bool WdbeQFilList::readXML(
			xmlXPathContext* docctx
			, string basexpath
			, bool addbasetag
		) {
	bool basefound;

	if (addbasetag)
		basefound = checkUclcXPaths(docctx, basexpath, basexpath, "WdbeQFilList");
	else
		basefound = checkXPath(docctx, basexpath);

	if (basefound) {
		extractStringUclc(docctx, basexpath, "stubGrp", "grp", stubGrp);
		extractStringUclc(docctx, basexpath, "stubOwn", "own", stubOwn);
		extractStringUclc(docctx, basexpath, "Filename", "fnm", Filename);
		extractStringUclc(docctx, basexpath, "srefRefIxVTbl", "ret", srefRefIxVTbl);
		extractStringUclc(docctx, basexpath, "titRefIxVTbl", "ret2", titRefIxVTbl);
		extractStringUclc(docctx, basexpath, "stubRefUref", "reu", stubRefUref);
		extractStringUclc(docctx, basexpath, "osrefKContent", "cnt", osrefKContent);
		extractStringUclc(docctx, basexpath, "titOsrefKContent", "cnt2", titOsrefKContent);
		extractStringUclc(docctx, basexpath, "srefKMimetype", "mim", srefKMimetype);
		extractStringUclc(docctx, basexpath, "titSrefKMimetype", "mim2", titSrefKMimetype);
		extractUsmallintUclc(docctx, basexpath, "Size", "siz", Size);
	};

	return basefound;
};

/******************************************************************************
 class ListWdbeQFilList
 ******************************************************************************/

ListWdbeQFilList::ListWdbeQFilList() {
};

ListWdbeQFilList::ListWdbeQFilList(
			const ListWdbeQFilList& src
		) {
	nodes.resize(src.nodes.size(), NULL);
	for (unsigned int i = 0; i < nodes.size(); i++) nodes[i] = new WdbeQFilList(*(src.nodes[i]));
};

ListWdbeQFilList& ListWdbeQFilList::operator=(
			const ListWdbeQFilList& src
		) {
	nodes.resize(src.nodes.size(), NULL);
	for (unsigned int i = 0; i < nodes.size(); i++) nodes[i] = new WdbeQFilList(*(src.nodes[i]));

	return *this;
};

ListWdbeQFilList::~ListWdbeQFilList() {
	clear();
};

void ListWdbeQFilList::clear() {
	for (unsigned int i = 0; i < nodes.size(); i++) if (nodes[i]) delete nodes[i];
	nodes.resize(0);
};

bool ListWdbeQFilList::readXML(
			xmlXPathContext* docctx
			, string basexpath
			, bool addbasetag
		) {
	bool basefound;

	vector<unsigned int> ics;
	vector<bool> shorttags;

	WdbeQFilList* rec = NULL;

	string s;

	if (addbasetag)
		basefound = checkUclcXPaths(docctx, basexpath, basexpath, "ListWdbeQFilList");
	else
		basefound = checkXPath(docctx, basexpath);

	extractList(docctx, basexpath, "WdbeQFilList", "row", "jnum", ics, shorttags);

	clear();

	for (unsigned int i = 0; i < ics.size(); i++) {
		rec = new WdbeQFilList(ics[i]);

		if (shorttags[i]) s = basexpath + "/row[@jnum='" + to_string(ics[i]) + "']";
		else s = basexpath + "/WdbeQFilList[@jnum='" + to_string(ics[i]) + "']";

		if (rec->readXML(docctx, s)) nodes.push_back(rec);
		else delete rec;
	};

	return basefound;
};
